### DISCUSSION TITLE
Woche 5 - Backend Entwicklung

### DISCUSSION THREAD

**Discussion by lukashert** [Chrissi-Ruege/HealthKeeper#27] [General]
Nachdem unser Fokus, in den letzten Wochen, vor allem auf der Entwicklung des Frontend lag, haben wir in dieser Woche am Backend gearbeitet. 

### EF Core
Wir möchten unsere Daten in einer SQL basierten Datenbank speichern. Um mit dem Backend darauf Zugreifen zu können, haben wir das Entity Framework (EF) eingebunden. Dabei handelt es sich um einen, von Microsoft entwickelten, ORM Mapper. 

### Identity System
Wir haben damit angefangen ein Identity System in unsere Application zu implementieren, durch dieses soll es später möglich sein eigene Benutzerkonten anzulegen und zu verwalten.

### Statistiken
Wir haben einen Controller zum Lesen und Schreiben von Statistikeinträgen. Ein Eintrag besteht dabei aus Gewicht, Größe und Zeitpunkt. Später soll es möglich sein diese Einträge im Frontend zu verwalten und dadurch die Entwicklung des eigenen Gewichts einsehen zu können.  Die Größe wird benötigt um aus dem Gewicht einen BMI zu berechnen.
Dieser Controller wird noch mit Unit Tests großflächig abgedeckt werden.

### Ausblick
In den nächsten Wochen möchten wir weiter an unserem Backend arbeiten. Ein Test Framework soll eingebunden werden und das Frontend soll Refactored werden.

**Comment by niklaskrauth** [comment]
Schön zu sehen das ihr wieder euren Fokus auf euer Backend legt.
